tests/mir-opt/slice_iter_next.rs

```diff
@@ -0,0 +1,24 @@
+// compile-flags: -O -Z inline-mir-hint-threshold=10000 -C debuginfo=0
```

Comment:
I presume with this set of flags you're looking to simulate a normal `--release` build. To do that in a mir-opt test you need to also pass `-Zmir-opt-level=2`, because mir-opt tests default you to `-Zmir-opt-level=4`.